packages/frontend/src/pages/StartRecipe.svelte

```diff
@@ -123,21 +123,23 @@ async function submit(): Promise<void> {
   if (!recipe || !value) return;
 
   loading = true;
-  trackingId = await studioClient.requestPullApplication({
+  const trackingId = await studioClient.requestPullApplication({
     recipeId: recipe.id,
     modelId: value.id,
     connection: containerProviderConnection,
   });
   router.location.query.set('trackingId', trackingId);
 }
 
-onMount(() => {
-  // Fetch any trackingId we could recover from query
-  const query = router.location.query.get('trackingId');
-  if (typeof query === 'string' && query.length > 0) {
-    trackingId = query;
-  }
+$: if (typeof trackingId === 'string' && trackingId.length > 0) {
+  refreshTasks();
+}
 
+function refreshTasks(): void {
```

Comment:
might be a good opportunity to rename refreshTasks to

$effect(() => {

});

so if trackingId being a property change, it'll do the updates

later on, tasks being a state, it would just need to have the effect method being called to process the tasks once trackId or tasks are being updated

Reply:
I opened https://github.com/containers/podman-desktop-extension-ai-lab/issues/1789

which would cover the better management of the states